File: ruby/word-count/word_count.rb
class Phrase

  def initialize(phrase)
    @words = phrase.downcase.scan(/\b[^\s,]+\b/).map
  end

  def word_count
    @words.uniq.map { |word| [word, @words.count(word)] }.to_h
  end

end

module BookKeeping
  VERSION = 1
end

File: python/word-count/word_count.py
import re

def word_count(phrase):
    word_counts = {}

    phrase_without_punctuation = re.sub(r"[.:!&@$%^&]", "", phrase)
    words = re.split(r'[,_\t\n\s]\s*', phrase_without_punctuation)

    for word in words:
        lowercase_word = word.lower()

        if lowercase_word in word_counts:
            word_counts[lowercase_word] += 1
        else:
            word_counts[lowercase_word] = 1

    return word_counts
